decam_process.py

```diff
@@ -15,324 +15,378 @@
-    via processCcd.
-    For successful difference imaging in the next step, astrometry and
-    photometric calibration must also be done by processCcd.
-    These steps need a reference catalog. The catalog used here is pan-starrs,
-    which lives on lsst-dev at /datasets/refcats/htm/ps1_pv3_3pi_20170110/
-    This catalog must exist in repo/ref_cats, e.g.,
-    $ ln -s /datasets/refcats/htm/ps1_pv3_3pi_20170110/ /path/to/repo/ref_cats
-    For more information, see RFC-257, DM-8232, and
-    https://community.lsst.org/t/creating-and-using-new-style-reference-catalogs
-    USAGE:
-    $ python hits_ingest.py processCcd
-    BASH EQUIVALENT:
-    $ processCcd.py repo --id visit=visit ccdnum=ccdnum
-            --output processedrepo --calib calibrepo
-            -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
-            -C processccd_config.py
-            --config calibrate.doAstrometry=True calibrate.doPhotoCal=True
-    ** to run from bash, 'processccd_config.py' must exist and contain, e.g.,
-        from lsst.meas.algorithms import LoadIndexedReferenceObjectsTask
-        for refObjLoader in (config.calibrate.astromRefObjLoader,
-                            config.calibrate.photoRefObjLoader,
-                            config.charImage.refObjLoader,
-                            ):
-        refObjLoader.retarget(LoadIndexedReferenceObjectsTask)
-        refObjLoader.ref_dataset_name = 'pan-starrs'
-        refObjLoader.filterMap = {"g": "g",
-                                  "r": "r",
-                                  "VR": "g"}
-    RESULT:
-    processedrepo/visit populated with subdirectories containing the
-    usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR).
-
-    diffIm - Do difference imaging with a visit as a template and
-    one or more as science.
-    USAGE:
-    $ python hits_ingest.py diffIm
-    BASH EQUIVALENT:
-    $ imageDifference.py processedrepo --id visit=sciencevisit ccdnum=ccdnum
-            --templateId visit=templatevisit --output diffimrepo
-            -C diffim_config.py
-    ** to run from bash, 'diffim_config.py' must exist and contain, e.g.,
-        from lsst.ip.diffim.getTemplate import GetCalexpAsTemplateTask
-        config.getTemplate.retarget(GetCalexpAsTemplateTask)
-        config.detection.thresholdValue=5.0
-        config.doDecorrelation=True
-    RESULT:
-    diffimrepo/deepDiff/v+sciencevisit populated with difference images
-    and catalogs of detected sources (diaSrc, diffexp, and metadata files)
-                        '''))
-    parser.add_argument('-f', '--files',
-                        help='Input files or directory for ingest or ingestCalibs.')
+    parser.add_argument('-d', '--dataset',
+                        help="Location on disk of dataset_root, which contains subdirectories of raw data, calibs, etc.")
+    parser.add_argument('-o', '--output',
+                        help="Location on disk where output repos will live.")
+    # TODO: implement this argument
```

Comment:
It is part of DM-11422 because using coadds as templates will require `--id`-like metadata to be handled differently than when a visit is a template. I'll note this explicitly here.